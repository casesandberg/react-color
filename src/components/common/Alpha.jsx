'use strict';

var React = require('react');
var ReactCSS = require('reactcss');

var Checkboard = require('./Checkboard');

class Alpha extends ReactCSS.Component {

  constructor() {
    super();

    this.handleChange = this.handleChange.bind(this);
    this.handleMouseDown = this.handleMouseDown.bind(this);
    this.handleMouseUp = this.handleMouseUp.bind(this);
  }

  classes() {
    return {
      'default': {
        alpha: {
          Absolute: '0 0 0 0',
          borderRadius: this.props.radius,
        },
        checkboard: {
          Absolute: '0 0 0 0',
          overflow: 'hidden',
        },
        gradient: {
          Absolute: '0 0 0 0',
          background: 'linear-gradient(to right, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 0) 0%, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 1) 100%)',
          boxShadow: this.props.shadow,
          borderRadius: this.props.radius,
        },
        container: {
          position: 'relative',
          zIndex: '2',
          height: '100%',
          margin: '0 3px',
        },
        pointer: {
          zIndex: '2',
          position: 'absolute',
          left: this.props.rgb.a * 100 + '%',
        },
        slider: {
          width: '4px',
          borderRadius: '1px',
          height: '8px',
          boxShadow: '0 0 2px rgba(0, 0, 0, .6)',
          background: '#fff',
          marginTop: '1px',
          transform: 'translateX(-2px)',
        },
      },
    };
  }

  handleChange(e, skip) {
    !skip && e.preventDefault();
    var container = this.refs.container;
    var containerWidth = container.clientWidth;
    var left = (e.pageX || e.touches[0].pageX) - (container.getBoundingClientRect().left + window.pageXOffset);

    var a;
    if (left < 0) {
      a = 0;
    } else if (left > containerWidth) {
      a = 1;
    } else {
      a = Math.round(left * 100 / containerWidth) / 100;
    }

    if (this.props.a !== a) {
      this.props.onChange({ h: this.props.hsl.h, s: this.props.hsl.s, l: this.props.hsl.l, a: a });
    }
  }

  handleMouseDown(e) {
    this.handleChange(e, true);
    window.addEventListener('mousemove', this.handleChange);
    window.addEventListener('mouseup', this.handleMouseUp);
  }

  handleMouseUp() {
    window.removeEventListener('mousemove', this.handleChange);
    window.removeEventListener('mouseup', this.handleMouseUp);
  }

  render() {
    var pointer = <div is="slider" />;

    if (this.props.pointer) {
      pointer = <this.props.pointer {...this.props} />;
    }

    return (
      <div is="alpha">
        <div is="checkboard">
          <Checkboard />
        </div>
        <div is="gradient" />
        <div is="container" ref="container" onMouseDown={ this.handleMouseDown } onTouchMove={ this.handleChange }>
          <div is="pointer" ref="pointer">
            { pointer }
          </div>
        </div>
      </div>
    );
  }
}

module.exports = Alpha;
