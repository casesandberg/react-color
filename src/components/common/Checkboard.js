'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

let _checkboardCache = {}

function renderCheckboard(c1: string, c2: string, size: number): any {
  if (typeof document == 'undefined') return null // Dont Render On Server
  var canvas: any = document.createElement('canvas')
  canvas.width = canvas.height = size * 2
  var ctx = canvas.getContext('2d')
  if (!ctx) return null // If no context can be found, return early.
  ctx.fillStyle = c1
  ctx.fillRect(0, 0, canvas.width, canvas.height)
  ctx.fillStyle = c2
  ctx.fillRect(0, 0, size, size)
  ctx.translate(size, size)
  ctx.fillRect(0, 0, size, size)
  return canvas.toDataURL()
}

function getCheckboard(c1: string, c2: string, size: number): any {
  var key = c1 + ',' + c2 + ',' + size

  if (_checkboardCache[key]) {
    return _checkboardCache[key]
  } else {
    var checkboard = renderCheckboard(c1, c2, size)
    _checkboardCache[key] = checkboard
    return checkboard
  }
}

export class Checkboard extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    var background = getCheckboard(this.props.white, this.props.grey, this.props.size)
    return {
      'default': {
        grid: {
          Absolute: '0px 0px 0px 0px',
          background: 'url(' + background + ') center left',
        },
      },
    }
  }

  render(): any {
    return (
      <div is="grid" ref="grid"></div>
    )
  }

}

Checkboard.defaultProps = {
  size: 8,
  white: '#fff',
  grey: '#e6e6e6',
}

export default Checkboard
