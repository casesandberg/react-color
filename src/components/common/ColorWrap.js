'use strict' /* @flow */

import React from 'react'
import merge from 'merge'
import isPlainObject from 'lodash/isplainobject'
import debounce from 'lodash/debounce'
import color from '../../helpers/color'
import shallowCompare from 'react-addons-shallow-compare'

export const ColorWrap = (Picker) => {
  class ColorPicker extends React.Component {
    shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1]);

    constructor(props: any) {
      super()

      this.state = merge(color.toState(props.color, 0), {
        visible: props.display,
      })

      this.debounce = debounce(function (fn: any, data: any) {
        fn(data)
      }, 100)
    }

    handleChange = (data: any) => {
      data = color.simpleCheckForValidColor(data)
      if (data) {
        var colors = color.toState(data, data.h || this.state.oldHue)
        this.setState(colors)
        this.props.onChangeComplete && this.debounce(this.props.onChangeComplete, colors)
        this.props.onChange && this.props.onChange(colors)
      }
    }

    componentWillReceiveProps(nextProps: any) {
      this.setState(merge(color.toState(nextProps.color, this.state.oldHue), {
        visible: nextProps.display,
      }))
    }

    render(): any {
      return <Picker {...this.props} {...this.state} onChange={ this.handleChange } />
    }
  }

  ColorPicker.defaultProps = {
    color: {
      h: 250,
      s: .50,
      l: .20,
      a: 1,
    },
  }

  return ColorPicker
}

export default ColorWrap
