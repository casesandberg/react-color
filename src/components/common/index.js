export { default as Alpha } from './Alpha'
export { default as Checkboard } from './Checkboard'
export { default as EditableInput } from './EditableInput'
export { default as Hue } from './Hue'
export { default as Raised } from './Raised'
export { default as Saturation } from './Saturation'
export { default as ColorWrap } from './ColorWrap'
export { default as Swatch } from './Swatch'
