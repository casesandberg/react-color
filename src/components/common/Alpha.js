'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import shallowCompare from 'react-addons-shallow-compare'

import Checkboard from './Checkboard'

export class Alpha extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    return {
      'default': {
        alpha: {
          Absolute: '0px 0px 0px 0px',
          borderRadius: this.props.radius,
        },
        checkboard: {
          Absolute: '0px 0px 0px 0px',
          overflow: 'hidden',
        },
        gradient: {
          Absolute: '0px 0px 0px 0px',
          background: 'linear-gradient(to right, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 0) 0%, rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', 1) 100%)',
          boxShadow: this.props.shadow,
          borderRadius: this.props.radius,
        },
        container: {
          position: 'relative',
          zIndex: '2',
          height: '100%',
          margin: '0px 3px',
        },
        pointer: {
          zIndex: '2',
          position: 'absolute',
          left: this.props.rgb.a * 100 + '%',
        },
        slider: {
          width: '4px',
          borderRadius: '1px',
          height: '8px',
          boxShadow: '0px 0px 2px rgba(0, 0, 0, .6)',
          background: '#fff',
          marginTop: '1px',
          transform: 'translateX(-2px)',
        },
      },
    }
  }

  componentWillUnmount() {
    this.unbindEventListeners()
  }

  handleChange = (e: any, skip: boolean) => {
    !skip && e.preventDefault()
    var container = this.refs.container
    var containerWidth = container.clientWidth
    var left = (e.pageX || e.touches[0].pageX) - (container.getBoundingClientRect().left + window.pageXOffset)

    var a
    if (left < 0) {
      a = 0
    } else if (left > containerWidth) {
      a = 1
    } else {
      a = Math.round(left * 100 / containerWidth) / 100
    }

    if (this.props.a !== a) {
      this.props.onChange({
        h: this.props.hsl.h,
        s: this.props.hsl.s,
        l: this.props.hsl.l,
        a: a,
        source: 'rgb',
      })
    }
  }

  handleMouseDown = (e: any) => {
    this.handleChange(e, true)
    window.addEventListener('mousemove', this.handleChange)
    window.addEventListener('mouseup', this.handleMouseUp)
  }

  handleMouseUp = () => {
    this.unbindEventListeners()
  }

  unbindEventListeners = () => {
    window.removeEventListener('mousemove', this.handleChange)
    window.removeEventListener('mouseup', this.handleMouseUp)
  }

  render(): any {
    var pointer = <div is="slider" />

    if (this.props.pointer) {
      pointer = <this.props.pointer {...this.props} />
    }

    return (
      <div is="alpha">
        <div is="checkboard">
          <Checkboard />
        </div>
        <div is="gradient" />
        <div is="container" ref="container" onMouseDown={ this.handleMouseDown }
            onTouchMove={ this.handleChange }
            onTouchStart={ this.handleChange }>
          <div is="pointer" ref="pointer">
            { pointer }
          </div>
        </div>
      </div>
    )
  }
}

export default Alpha
