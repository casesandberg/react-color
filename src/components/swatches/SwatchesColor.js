import React from 'react'
import reactCSS from 'reactcss'
import colorUtils from '../../helpers/color'

import { Swatch } from '../common'

export const SwatchesColor = ({ color, onClick = () => {}, onSwatchHover, first,
  last, active }) => {
  const styles = reactCSS({
    'default': {
      color: {
        width: '40px',
        height: '24px',
        cursor: 'pointer',
        background: color,
        marginBottom: '1px',
      },
      check: {
        fill: colorUtils.getContrastingColor(color),
        marginLeft: '8px',
        display: 'none',
      },
    },
    'first': {
      color: {
        overflow: 'hidden',
        borderRadius: '2px 2px 0 0',
      },
    },
    'last': {
      color: {
        overflow: 'hidden',
        borderRadius: '0 0 2px 2px',
      },
    },
    'active': {
      check: {
        display: 'block',
      },
    },
    'color-#FFFFFF': {
      color: {
        boxShadow: 'inset 0 0 0 1px #ddd',
      },
      check: {
        fill: '#333',
      },
    },
    'transparent': {
      check: {
        fill: '#333',
      },
    },
  }, {
    first,
    last,
    active,
    'color-#FFFFFF': color === '#FFFFFF',
    'transparent': color === 'transparent',
  })

  return (
    <Swatch
      color={ color }
      style={ styles.color }
      onClick={ onClick }
      onHover={ onSwatchHover }
      focusStyle={{ boxShadow: `0 0 4px ${ color }` }}
    >
      <div style={ styles.check }>
        <svg style={{ width: '24px', height: '24px' }} viewBox="0 0 24 24">
          <path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z" />
        </svg>
      </div>
    </Swatch>
  )
}

export default SwatchesColor
