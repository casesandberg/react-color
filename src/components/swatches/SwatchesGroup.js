import React from 'react'
import reactCSS from 'reactcss'

import SwatchesColor from './SwatchesColor'

export const SwatchesGroup = ({ onClick, onSwatchHover, group, active }) => {
  const styles = reactCSS({
    'default': {
      group: {
        paddingBottom: '10px',
        width: '40px',
        float: 'left',
        marginRight: '10px',
      },
    },
  })

  return (
    <div style={ styles.group }>
      { group.map((color, i) => (
        <SwatchesColor
          key={ color }
          color={ color }
          active={ color.toLowerCase() === active }
          first={ i === 0 }
          last={ i === group.length - 1 }
          onClick={ onClick }
          onSwatchHover={ onSwatchHover }
        />
      )) }
    </div>
  )
}

export default SwatchesGroup
