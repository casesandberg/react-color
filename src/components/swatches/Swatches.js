'use strict' /* @flow */

import React from 'react'
import ReactCSS from 'reactcss'
import color from '../../helpers/color'
import material from 'material-colors'
import shallowCompare from 'react-addons-shallow-compare'

import { ColorWrap } from '../common'
import { Raised } from '../../../modules/react-material-design'
import SwatchesGroup from './SwatchesGroup'

export class Swatches extends ReactCSS.Component {
  shouldComponentUpdate = shallowCompare.bind(this, this, arguments[0], arguments[1])

  classes(): any {
    return {
      'default': {
        picker: {
          width: this.props.width,
          height: this.props.height,
        },
        overflow: {
          height: this.props.height,
          overflowY: 'scroll',
        },
        body: {
          padding: '16px 0px 6px 16px',
        },

        clear: {
          clear: 'both',
        },
      },
    }
  }

  handleChange = (data: any) => {
    color.isValidHex(data) && this.props.onChange({
      hex: data,
      source: 'hex',
    })
  }

  render(): any {
    var groups = []
    if (this.props.colors) {
      for (var i = 0; i < this.props.colors.length; i++) {
        var group = this.props.colors[i]
        groups.push(<SwatchesGroup key={ group.toString() } group={ group } active={ this.props.hex } onClick={ this.handleChange } />)
      }
    }

    return (
      <div is="picker">
        <Raised>
          <div is="overflow">
            <div is="body" ref="body">
              { groups }
              <div is="clear" />
            </div>
          </div>
        </Raised>
      </div>
    )
  }
}

Swatches.defaultProps = {
  width: 320,
  height: 240,
  colors: [
    [material.red['900'], material.red['700'], material.red['500'], material.red['300'], material.red['100']],
    [material.pink['900'], material.pink['700'], material.pink['500'], material.pink['300'], material.pink['100']],
    [material.purple['900'], material.purple['700'], material.purple['500'], material.purple['300'], material.purple['100']],
    [material.deepPurple['900'], material.deepPurple['700'], material.deepPurple['500'], material.deepPurple['300'], material.deepPurple['100']],
    [material.indigo['900'], material.indigo['700'], material.indigo['500'], material.indigo['300'], material.indigo['100']],
    [material.blue['900'], material.blue['700'], material.blue['500'], material.blue['300'], material.blue['100']],
    [material.lightBlue['900'], material.lightBlue['700'], material.lightBlue['500'], material.lightBlue['300'], material.lightBlue['100']],
    [material.cyan['900'], material.cyan['700'], material.cyan['500'], material.cyan['300'], material.cyan['100']],
    [material.teal['900'],  material.teal['700'], material.teal['500'], material.teal['300'], material.teal['100']],
    ['#194D33', material.green['700'], material.green['500'], material.green['300'], material.green['100']],
    [material.lightGreen['900'], material.lightGreen['700'], material.lightGreen['500'], material.lightGreen['300'], material.lightGreen['100']],
    [material.lime['900'], material.lime['700'], material.lime['500'], material.lime['300'], material.lime['100']],
    [material.yellow['900'], material.yellow['700'], material.yellow['500'], material.yellow['300'], material.yellow['100']],
    [material.amber['900'], material.amber['700'], material.amber['500'], material.amber['300'], material.amber['100']],
    [material.orange['900'], material.orange['700'], material.orange['500'], material.orange['300'], material.orange['100']],
    [material.deepOrange['900'], material.deepOrange['700'], material.deepOrange['500'], material.deepOrange['300'], material.deepOrange['100']],
    [material.brown['900'], material.brown['700'], material.brown['500'], material.brown['300'], material.brown['100']],
    [material.blueGrey['900'], material.blueGrey['700'], material.blueGrey['500'], material.blueGrey['300'], material.blueGrey['100']],
  ],
}

export default ColorWrap(Swatches)
