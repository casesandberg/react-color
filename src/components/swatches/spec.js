/* global test, expect */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme';
import color, { red } from '../../helpers/color'

import Swatches from './Swatches'
import SwatchesColor from './SwatchesColor'
import SwatchesGroup from './SwatchesGroup'
import { Swatch } from '../common'

test('Swatches renders correctly', () => {
  const tree = renderer.create(
    <Swatches hex={ red.hex } colors={ [['#fff'], ['#333']] } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Swatches onChange events correctly', () => {
  const changeSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy()
  })
  const tree = mount(
    <Swatches onChange={changeSpy} />
  )
  expect(changeSpy).toHaveBeenCalledTimes(0)
  const swatches = tree.find(Swatch);
  swatches.at(0).childAt(0).simulate('click')

  expect(changeSpy).toHaveBeenCalled()
})

test('Swatches with onSwatchHover events correctly', () => {
  const hoverSpy = jest.fn((data) => {
    expect(color.simpleCheckForValidColor(data)).toBeTruthy()
  })
  const tree = mount(
    <Swatches onSwatchHover={hoverSpy} />
  )
  expect(hoverSpy).toHaveBeenCalledTimes(0)
  const swatches = tree.find(Swatch);
  swatches.at(0).childAt(0).simulate('mouseOver')

  expect(hoverSpy).toHaveBeenCalled()
})

test('SwatchesColor renders correctly', () => {
  const tree = renderer.create(
    <SwatchesColor />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SwatchesColor renders with props', () => {
  const tree = renderer.create(
    <SwatchesColor active first last />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SwatchesGroup renders correctly', () => {
  const tree = renderer.create(
    <SwatchesGroup active={ red.hex } group={ ['#fff'] } />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
