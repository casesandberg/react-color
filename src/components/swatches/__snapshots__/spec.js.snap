// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swatches renders correctly 1`] = `
<div
  className="swatches-picker "
  style={
    Object {
      "height": 240,
      "width": 320,
    }
  }
>
  <div
    style={
      Object {
        "display": "inline-block",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "MozBorderRadius": 2,
          "MozBoxShadow": "0 2px 10px rgba(0,0,0,.12), 0 2px 5px rgba(0,0,0,.16)",
          "OBorderRadius": 2,
          "OBoxShadow": "0 2px 10px rgba(0,0,0,.12), 0 2px 5px rgba(0,0,0,.16)",
          "WebkitBorderRadius": 2,
          "WebkitBoxShadow": "0 2px 10px rgba(0,0,0,.12), 0 2px 5px rgba(0,0,0,.16)",
          "background": "#fff",
          "borderRadius": 2,
          "bottom": "0px",
          "boxShadow": "0 2px 10px rgba(0,0,0,.12), 0 2px 5px rgba(0,0,0,.16)",
          "left": "0px",
          "msBorderRadius": 2,
          "msBoxShadow": "0 2px 10px rgba(0,0,0,.12), 0 2px 5px rgba(0,0,0,.16)",
          "position": "absolute",
          "right": "0px",
          "top": "0px",
        }
      }
    />
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "height": 240,
            "overflowY": "scroll",
          }
        }
      >
        <div
          style={
            Object {
              "padding": "16px 0 6px 16px",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "marginRight": "10px",
                "paddingBottom": "10px",
                "width": "40px",
              }
            }
          >
            <span
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <div
                onClick={[Function]}
                onKeyDown={[Function]}
                style={
                  Object {
                    "MozBorderRadius": "0 0 2px 2px",
                    "OBorderRadius": "0 0 2px 2px",
                    "WebkitBorderRadius": "0 0 2px 2px",
                    "background": "#fff",
                    "borderRadius": "0 0 2px 2px",
                    "cursor": "pointer",
                    "height": "24px",
                    "marginBottom": "1px",
                    "msBorderRadius": "0 0 2px 2px",
                    "outline": "none",
                    "overflow": "hidden",
                    "position": "relative",
                    "width": "40px",
                  }
                }
                tabIndex={0}
                title="#fff"
              >
                <div
                  style={
                    Object {
                      "display": "none",
                      "fill": "#000",
                      "marginLeft": "8px",
                    }
                  }
                >
                  <svg
                    style={
                      Object {
                        "height": "24px",
                        "width": "24px",
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"
                    />
                  </svg>
                </div>
              </div>
            </span>
          </div>
          <div
            style={
              Object {
                "float": "left",
                "marginRight": "10px",
                "paddingBottom": "10px",
                "width": "40px",
              }
            }
          >
            <span
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <div
                onClick={[Function]}
                onKeyDown={[Function]}
                style={
                  Object {
                    "MozBorderRadius": "0 0 2px 2px",
                    "OBorderRadius": "0 0 2px 2px",
                    "WebkitBorderRadius": "0 0 2px 2px",
                    "background": "#333",
                    "borderRadius": "0 0 2px 2px",
                    "cursor": "pointer",
                    "height": "24px",
                    "marginBottom": "1px",
                    "msBorderRadius": "0 0 2px 2px",
                    "outline": "none",
                    "overflow": "hidden",
                    "position": "relative",
                    "width": "40px",
                  }
                }
                tabIndex={0}
                title="#333"
              >
                <div
                  style={
                    Object {
                      "display": "none",
                      "fill": "#fff",
                      "marginLeft": "8px",
                    }
                  }
                >
                  <svg
                    style={
                      Object {
                        "height": "24px",
                        "width": "24px",
                      }
                    }
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"
                    />
                  </svg>
                </div>
              </div>
            </span>
          </div>
          <div
            style={
              Object {
                "clear": "both",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SwatchesColor renders correctly 1`] = `
<span
  onBlur={[Function]}
  onFocus={[Function]}
>
  <div
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "background": undefined,
        "cursor": "pointer",
        "height": "24px",
        "marginBottom": "1px",
        "outline": "none",
        "position": "relative",
        "width": "40px",
      }
    }
    tabIndex={0}
    title={undefined}
  >
    <div
      style={
        Object {
          "display": "none",
          "fill": "#fff",
          "marginLeft": "8px",
        }
      }
    >
      <svg
        style={
          Object {
            "height": "24px",
            "width": "24px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"
        />
      </svg>
    </div>
  </div>
</span>
`;

exports[`SwatchesColor renders with props 1`] = `
<span
  onBlur={[Function]}
  onFocus={[Function]}
>
  <div
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "MozBorderRadius": "0 0 2px 2px",
        "OBorderRadius": "0 0 2px 2px",
        "WebkitBorderRadius": "0 0 2px 2px",
        "background": undefined,
        "borderRadius": "0 0 2px 2px",
        "cursor": "pointer",
        "height": "24px",
        "marginBottom": "1px",
        "msBorderRadius": "0 0 2px 2px",
        "outline": "none",
        "overflow": "hidden",
        "position": "relative",
        "width": "40px",
      }
    }
    tabIndex={0}
    title={undefined}
  >
    <div
      style={
        Object {
          "display": "block",
          "fill": "#fff",
          "marginLeft": "8px",
        }
      }
    >
      <svg
        style={
          Object {
            "height": "24px",
            "width": "24px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"
        />
      </svg>
    </div>
  </div>
</span>
`;

exports[`SwatchesGroup renders correctly 1`] = `
<div
  style={
    Object {
      "float": "left",
      "marginRight": "10px",
      "paddingBottom": "10px",
      "width": "40px",
    }
  }
>
  <span
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "MozBorderRadius": "0 0 2px 2px",
          "OBorderRadius": "0 0 2px 2px",
          "WebkitBorderRadius": "0 0 2px 2px",
          "background": "#fff",
          "borderRadius": "0 0 2px 2px",
          "cursor": "pointer",
          "height": "24px",
          "marginBottom": "1px",
          "msBorderRadius": "0 0 2px 2px",
          "outline": "none",
          "overflow": "hidden",
          "position": "relative",
          "width": "40px",
        }
      }
      tabIndex={0}
      title="#fff"
    >
      <div
        style={
          Object {
            "display": "none",
            "fill": "#000",
            "marginLeft": "8px",
          }
        }
      >
        <svg
          style={
            Object {
              "height": "24px",
              "width": "24px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"
          />
        </svg>
      </div>
    </div>
  </span>
</div>
`;
